### PR TITLE
Few more LambdaUI fixes

### PR DESCRIPTION
**Pull request by rakesh-garimella**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [ ] I have updated the relevant documentation.
---

**While creating a new Lambda unless lambda name is provided user cannot interact with other components on lambda page**

Changes proposed in this pull request:

- Now unless the lambda name is not entered
   user cannot interact with other components

### CODE REVIEW

File: lambda/src/app/lambdas/lambda-details/lambda-env/lambda-env.component.html [hunk 1]
```diff
@@ -20,7 +20,7 @@ <h2> Environment Variable </h2>
             <input [(ngModel)]="envVar.value" class="tn-form__control" type="text" id="input-2" placeholder="Value" (keyup.enter)="addVariable()">
```
Comment (sayanh) [inline lambda/src/app/lambdas/lambda-details/lambda-env/lambda-env.component.html]: IMO input fields for `VARIABLE_NAME` and `Value` should be disabled as well

**Comment by sayanh** [comment]
I think this [PR](https://github.com/kyma-project/console/pull/15) uses variable `isLambdaFormValid` to exactly tackle the problem we are solving here. May be we can consolidate both.

File: lambda/src/app/lambdas/lambda-details/lambda-details.component.html [hunk 2]
```diff
@@ -138,15 +139,15 @@ <h2> Code </h2>
                 <div class="triggers-dropdown__close" (click)="toggleDropdown($event)">
                   <div class="icon tn-icon tn-icon--remove" dropdown="trigger" (click)="hideEventDropdown()"></div>
                 </div>
-                <app-event-trigger-chooser [availableTriggers]="availableEventTriggers" [selectedTriggers]="selectedTriggers"></app-event-trigger-chooser>
+                <app-event-trigger-chooser [availableTriggers]="availableEventTriggers" [lambdaName]="lambda.metadata.name" [isLambdaNameInvalid]="isFunctionNameInvalid" [selectedTriggers]="selectedTriggers"></app-event-trigger-chooser>
```
Comment (sayanh) [inline lambda/src/app/lambdas/lambda-details/lambda-details.component.html]: As the dropdown for `Event Trigger` is disabled, why do we need to send `isLambdaNameInvalid` to `EventTriggerChooserComponent`?

File: lambda/src/app/lambdas/lambda-details/lambda-instance-bindings/lambda-instance-bindings.component.html [hunk 3]
```diff
@@ -25,7 +25,7 @@ <h2> Service Instance Bindings </h2>
       </div>
     </div>
     <div class="col-1">
-      <button class="tn-icon tn-icon--delete tn-button tn-button--small tn-button--icon tn-button--text" (click)="remove(i)"></button>
+      <button class="tn-icon tn-icon--delete tn-button tn-button--small tn-button--icon tn-button--text" (click)="remove(i)" [disabled]="!lambdaName || isLambdaNameInvalid"></button>
```
Comment (sayanh) [inline lambda/src/app/lambdas/lambda-details/lambda-instance-bindings/lambda-instance-bindings.component.html]: Again disabling this button is redundant when button `+ Add Service Instance Binding` is disabled

File: lambda/src/app/lambdas/lambda-details/lambda-env/lambda-env.component.html (hunk 1 above)
Comment (rakesh-garimella) [reply lambda/src/app/lambdas/lambda-details/lambda-env/lambda-env.component.html]: done.

File: lambda/src/app/lambdas/lambda-details/lambda-details.component.ts [hunk 4]
```diff
@@ -854,13 +854,13 @@ export class LambdaDetailsComponent implements AfterViewInit {
...
         : true;
-    !this.lambda.metadata.name || this.isFunctionNameInvalid
-      ? this.editor.setReadOnly(true)
-      : this.editor.setReadOnly(false);
-
-    this.isFunctionNameInvalid
-      ? (this.isLambdaFormValid = false)
-      : (this.isLambdaFormValid = true);
+    if (!this.lambda.metadata.name || this.isFunctionNameInvalid) {
+      this.editor.setReadOnly(true);
```
Comment (sayanh) [inline lambda/src/app/lambdas/lambda-details/lambda-details.component.ts]: `this.dependencyEditor.setReadOnly(true);` is also needed

**Pull request merged**